### ISSUE TITLE
Error processing tar file(duplicates of file paths not supported)

### ISSUE THREAD

**Issue by gboutry** [canonical/rockcraft#203]
Hi,

Trying to pack a ROCK image for the first time and encountering issues.

I'm trying to pack a Keystone image with the following `rockcraft.yaml`:

```yaml
name: keystone
summary: Openstack Keystone
license: Apache-2.0
description: |
  Ubuntu distribution of OpenStack Keystone
version: antelope

base: ubuntu:22.04
platforms:
  amd64:

parts:
  cloud-archive:
    plugin: nil
    build-packages:
      - ubuntu-cloud-keyring
      - software-properties-common
    override-pull: |
      craftctl default
      add-apt-repository cloud-archive:antelope
      apt update

  keystone:
    after: [cloud-archive]
    plugin: nil
    stage-packages:
      - apache2
      - keystone
```

When I try to import the generated ROCK image using `skopeo`:
```
skopeo copy --insecure-policy oci-archive:keystone_antelope_amd64.rock docker-daemon:keystone:antelope
Getting image source signatures
Copying blob 677076032cca done  
Copying blob 4af93843b91c done  
Copying blob 4bfd57fdda04 done  
Copying config 05d3bd154c done  
Writing manifest to image destination
Storing signatures
```

It's missing from docker registry:
```bash
docker images
REPOSITORY    TAG        IMAGE ID       CREATED         SIZE
hello-world   latest     feb5d9fea6a5   17 months ago   13.3kB
```

When uploading to local registry with `skopeo` then downloading with `docker`:
```bash
docker pull localhost:32000/keystone:antelope                                                                                            
antelope: Pulling from keystone
677076032cca: Already exists 
4af93843b91c: Extracting [==================================================>]  9.225MB/9.225MB
4bfd57fdda04: Download complete 
failed to register layer: Error processing tar file(duplicates of file paths not supported):
```

When following the procedure [here](https://github.com/canonical/rockcraft/issues/194):
```bash
skopeo copy --insecure-policy oci-archive:keystone_antelope_amd64.rock oci:keystone
❯ cd keystone 
❯ manifest=$(jq -r '.manifests[].digest' < index.json | cut -d ':' -f 2)
❯ cd blobs/sha256 
❯ blob_name=$(jq -r '.layers[1].digest' < $manifest | cut -d ':' -f 2)
❯ tar -tvf $blob_name | awk -F' ' '{print $6}' | sort | uniq -d       
usr/lib/x86_64-linux-gnu/
```

How do we handle duplicate paths in generated rocks ?

**Comment by tigarmo** [comment]
Hi, can you paste the output of `snap info rockcraft`?

**Comment by gboutry** [comment]
```bash
snap info rockcraft
name:      rockcraft
summary:   A craft like experience to create ROCKS
publisher: Sergio Schvezov ⭐ (sergiusens)
store-url: https://snapcraft.io/rockcraft
license:   GPL-3.0
description: |
  Rockcraft aims to take the same primitives used in Charmcraft and Snapcraft
  to create OCI images.
commands:
  - rockcraft
snap-id:      3mPIBn6L9IYLyBAZsaqjlnMk7huEtqlv
tracking:     latest/edge
refresh-date: today at 09:47 CET
channels:
  latest/stable:    –                                   
  latest/candidate: –                                   
  latest/beta:      –                                   
  latest/edge:      0+git.3f43e30 2023-02-24 (613) 83MB classic
installed:          0+git.3f43e30            (613) 83MB classic
```

**Comment by tigarmo** [comment]
Thanks

> How do we handle duplicate paths in generated rocks ?

We don't, currently. It's a bug so now I'm thinking about the fix. In your case, the issue is that your ROCK has both `/lib/x86_64-linux-gnu/` and `/usr/lib/x86_64-linux-gnu/`, but because the base is ubuntu `/lib` is ultimately a symlink to `/usr/lib`, so there are indeed two directories that become `/usr/lib/x86_64-linux-gnu/`, as your original post indicates.

I'm thinking we can try to merge the directories, and fail if they have conflicting files (possibly just use craft-parts' merge handling for this).

**Comment by gboutry** [comment]
> I'm thinking we can try to merge the directories, and fail if they have conflicting files (possibly just use craft-parts' merge handling for this).

Is this something I'm supposed to do in my `rockcraft.yaml` or is this coming into `rockcraft` ?

**Comment by tigarmo** [comment]
Oh this is a bugfix in `rockcraft` itself.
